x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/SumDoubleAggregator.java

```diff
@@ -116,7 +119,12 @@ public static Block evaluateFinal(GroupingSumState state, IntVector selected, Dr
             for (int i = 0; i < selected.getPositionCount(); i++) {
                 int si = selected.getInt(i);
                 if (state.hasValue(si) && si < state.values.size()) {
-                    builder.appendDouble(state.values.get(si));
+                    var value = state.values.get(si);
+                    if (Double.isFinite(value) == false) {
+                        builder.appendNull();
+                    } else {
+                        builder.appendDouble(state.values.get(si));
+                    }
```

Comment:
I wonder if this is something we should push to the DoubleArrayState somehow. We don't want to push it on all the reads, but maybe some kind of `finalRead` or something? I dunno. I think `sum` is a lot more likely to get to Infinity than most aggs given that we'll never be sending in `Infinity`